Disabling BWC tests does not disable BWC build

This PR build failed: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request/2024/console

This build is on the `zen2` branch, on which BWC tests are currently disabled:

```
========================= WARNING =========================
         Backwards compatibility tests are disabled!
See https://github.com/elastic/elasticsearch/issues/32006
===========================================================
```

Nonetheless the build failed when trying to build the `next-bugfix-snapshot` version, which I think shouldn't be necessary if the BWC tests are disabled.

```
> Task :distribution:bwc:next-bugfix-snapshot:buildBwcVersion FAILED
FAILURE: Build failed with an exception.

:distribution:bwc:next-bugfix-snapshot:buildBwcVersion (Thread[Task worker for ':' Thread 10,5,main]) completed. Took 2 mins 7.955 secs.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/4.10/userguide/command_line_interface.html#sec:command_line_warnings
3068 actionable tasks: 1617 executed, 1451 up-to-date
* Where:
Build file '/var/lib/jenkins/workspace/elastic+elasticsearch+pull-request/distribution/bwc/build.gradle' line: 207

* What went wrong:
Execution failed for task ':distribution:bwc:next-bugfix-snapshot:buildBwcVersion'.
> Building bwc version didn't generate expected files [/var/lib/jenkins/workspace/elastic+elasticsearch+pull-request/distribution/bwc/next-bugfix-snapshot/build/bwc/checkout-6.4/distribution/archives/oss-zip/build/distributions/elasticsearch-oss-6.4.3-SNAPSHOT.zip, /var/lib/jenkins/workspace/elastic+elasticsearch+pull-request/distribution/bwc/next-bugfix-snapshot/build/bwc/checkout-6.4/distribution/archives/zip/build/distributions/elasticsearch-6.4.3-SNAPSHOT.zip, /var/lib/jenkins/workspace/elastic+elasticsearch+pull-request/distribution/bwc/next-bugfix-snapshot/build/bwc/checkout-6.4/distribution/packages/oss-deb/build/distributions/elasticsearch-oss-6.4.3-SNAPSHOT.deb, /var/lib/jenkins/workspace/elastic+elasticsearch+pull-request/distribution/bwc/next-bugfix-snapshot/build/bwc/checkout-6.4/distribution/packages/deb/build/distributions/elasticsearch-6.4.3-SNAPSHOT.deb, /var/lib/jenkins/workspace/elastic+elasticsearch+pull-request/distribution/bwc/next-bugfix-snapshot/build/bwc/checkout-6.4/distribution/packages/oss-rpm/build/distributions/elasticsearch-oss-6.4.3-SNAPSHOT.rpm, /var/lib/jenkins/workspace/elastic+elasticsearch+pull-request/distribution/bwc/next-bugfix-snapshot/build/bwc/checkout-6.4/distribution/packages/rpm/build/distributions/elasticsearch-6.4.3-SNAPSHOT.rpm]
```

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy
@@ -191,26 +191,32 @@ class VagrantTestPlugin implements Plugin<Project> {
             dependencies.add(project.dependencies.project(path: ":distribution:${it}", configuration: 'default'))
         }
 
-        // The version of elasticsearch that we upgrade *from*
-        VersionCollection.UnreleasedVersionInfo unreleasedInfo = project.bwcVersions.unreleasedInfo(upgradeFromVersion)
-        if (unreleasedInfo != null) {
-            // handle snapshots pointing to bwc build
-            UPGRADE_FROM_ARCHIVES.each {
-                dependencies.add(project.dependencies.project(
-                        path: ":distribution:bwc:${unreleasedInfo.gradleProjectName}", configuration: it))
-                if (upgradeFromVersion.onOrAfter('6.3.0')) {
+        if (project.ext.bwc_tests_enabled) {
+            // The version of elasticsearch that we upgrade *from*
+            // we only add them as dependencies if the bwc tests are enabled, so we don't trigger builds otherwise
+            VersionCollection.UnreleasedVersionInfo unreleasedInfo = project.bwcVersions.unreleasedInfo(upgradeFromVersion)
+            if (unreleasedInfo != null) {
+                // handle snapshots pointing to bwc build
+                UPGRADE_FROM_ARCHIVES.each {
                     dependencies.add(project.dependencies.project(
-                            path: ":distribution:bwc:${unreleasedInfo.gradleProjectName}", configuration: "oss-${it}"))
+                            path: ":distribution:bwc:${unreleasedInfo.gradleProjectName}", configuration: it))
+                    if (upgradeFromVersion.onOrAfter('6.3.0')) {
+                        dependencies.add(project.dependencies.project(
+                                path: ":distribution:bwc:${unreleasedInfo.gradleProjectName}", configuration: "oss-${it}"))
+                    }
+                }
+            } else {
+                UPGRADE_FROM_ARCHIVES.each {
+                    // The version of elasticsearch that we upgrade *from*
+                    dependencies.add("downloads.${it}:elasticsearch:${upgradeFromVersion}@${it}")
+                    if (upgradeFromVersion.onOrAfter('6.3.0')) {
+                        dependencies.add("downloads.${it}:elasticsearch-oss:${upgradeFromVersion}@${it}")
+                    }
                 }
             }
         } else {
-            UPGRADE_FROM_ARCHIVES.each {
-                // The version of elasticsearch that we upgrade *from*
-                dependencies.add("downloads.${it}:elasticsearch:${upgradeFromVersion}@${it}")
-                if (upgradeFromVersion.onOrAfter('6.3.0')) {
-                    dependencies.add("downloads.${it}:elasticsearch-oss:${upgradeFromVersion}@${it}")
-                }
-            }
+            // Upgrade tests will go from current to current when the BWC tests are disabled to skip real BWC tests.
+            upgradeFromVersion = Version.fromString(project.version)
         }
 
         for (Object dependency : dependencies) {

